docs/docs/developers/guides/smart_contracts/writing_contracts/how_to_emit_event.md

```diff
@@ -5,32 +5,39 @@ tags: [contracts]
 ---
 
 Events in Aztec work similarly to Ethereum events in the sense that they are a way for contracts to communicate with the outside world.
-They are emitted by contracts and stored inside each instance of an AztecNode.
 
-:::info
-Aztec events are currently represented as raw data and are not ABI encoded.
-ABI encoded events are a feature that will be added in the future.
-:::
+Events are structured pieces of data that can be emitted privately, from private functions, or publicly, from public functions. They include metadata about the event type, so people can query events to look for specific information.
+
+There are also public logs, which are similar to events, but are unstructured data.
+
+## Private Events
+
+To emit encrypted logs you can import the `encode_and_encrypt_event` or `encode_and_encrypt_event_unconstrained` functions and pass them into the `emit` function. An example can be seen in the reference token contract's transfer function:
+
+#include_code encrypted_unconstrained /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
 
-Unlike on Ethereum, there are 2 types of events supported by Aztec: [encrypted](#encrypted-events) and [unencrypted](#unencrypted-events).
+- `encode_and_encrypt_event` Sends an encrypted message to `recipient` with the content of the event, which they will discover when processing private logs.
+- `encode_and_encrypt_event_unconstrained` is the same as `encode_and_encrypt_event`, except encryption is unconstrained. This means that the sender is free to make the log contents be whatever they wish, potentially resulting in scenarios in which the recipient is unable to decrypt and process the payload, **leading to the event being lost**. Only use this function in scenarios where the recipient not receiving the event is an acceptable outcome.
```

Comment:
Curious if the unconstrained encryption also means that event data can be spoofed by the encryption process changing the data it's encrypting. If so, it would be important to highlight.

Reply:
it's not constrained, so in my mind it could be anything. The recipient has to trust the person they are getting the event from. I'll make a note that clarifies this